ee/vellum_ee/workflows/tests/test_server.py

```diff
@@ -566,3 +567,102 @@ class Outputs(BaseWorkflow.Outputs):
+    """
+    Test that resolve_workflow_deployment returns a workflow with artifacts
+    in generated_files using the correct prefix format.
+    """
+    # GIVEN a deployment name and release tag
+    deployment_name = "test_deployment"
+    release_tag = "v1.0.0"
+
+    expected_hash = str(uuid4_from_hash(f"{deployment_name}|{release_tag}")).replace("-", "_")
+    expected_prefix = f"vellum_workflow_deployment_{expected_hash}"
+
+    # Create a simple test node for the resolved workflow
+    test_node_code = """
+from vellum.workflows.nodes.bases.base import BaseNode
+from vellum.workflows.outputs import BaseOutputs
+
+class TestNode(BaseNode):
+    template = "Hello"
+
+    class Outputs(BaseOutputs):
+        result: str
+
+    def run(self):
+        return self.Outputs(result="Hello, {template}")
+"""
+
+    mock_workflow_code = """
+from vellum.workflows import BaseWorkflow
+from .nodes.test_node import TestNode
+
+class ResolvedWorkflow(BaseWorkflow):
+    graph = TestNode
+"""
+
+    # Create parent workflow files that reference the subworkflow deployment
+    parent_workflow_code = """
+from vellum.workflows import BaseWorkflow
+from .nodes.subworkflow_deployment_node import TestSubworkflowDeploymentNode
+
+class ParentWorkflow(BaseWorkflow):
+    graph = TestSubworkflowDeploymentNode
+"""
+
+    parent_node_code = """
+from vellum.workflows.nodes import SubworkflowDeploymentNode
+from vellum.workflows.outputs import BaseOutputs
+
+class TestSubworkflowDeploymentNode(SubworkflowDeploymentNode):
+    deployment = "test_deployment"
+
+    class Outputs(BaseOutputs):
+        result: str
+
+    subworkflow_inputs = {"message": "test"}
+"""
+
+    files = {
+        "__init__.py": "",
+        "workflow.py": parent_workflow_code,
+        "nodes/__init__.py": """
+from .subworkflow_deployment_node import TestSubworkflowDeploymentNode
+
+__all__ = ["TestSubworkflowDeploymentNode"]
+""",
+        "nodes/subworkflow_deployment_node.py": parent_node_code,
+        f"{expected_prefix}/workflow.py": mock_workflow_code,
+        f"{expected_prefix}/nodes/__init__.py": """
+from .test_node import TestNode
+
+__all__ = ["TestNode"]
+""",
+        f"{expected_prefix}/nodes/test_node.py": test_node_code,
+    }
+
+    namespace = str(uuid4())
+
+    # AND the virtual file loader is registered
+    finder = VirtualFileFinder(files, namespace)
+    sys.meta_path.append(finder)
+
+    try:
+        # WHEN we execute the root workflow
+        Workflow = BaseWorkflow.load_from_module(namespace)
+        workflow = Workflow(context=WorkflowContext(generated_files=files))
+
+        # THEN the workflow should be successfully initialized
+        assert workflow
+
+        event = workflow.run()
+
+        # AND the method should return a workflow (not None) - this will fail until implemented
+        assert event.name == "workflow.execution.rejected"
+        assert "Invalid API key" in str(event.error)
```

Comment:
switch these assertions to expecting fulfillment and add a `pytest.mark.skip` annotation to the test